io-engine/src/core/env.rs

```diff
@@ -477,6 +482,9 @@ pub fn mayastor_env_stop(rc: i32) {
             r.send_future(async move {
                 do_shutdown(rc as *const i32 as *mut c_void).await;
             });
+            MayastorEnvironment::global_or_default()
+                .event(EventAction::Stop)
+                .generate();
```

Comment:
Shouldn't this be generated right before we stop only?
And would it be worth it to include how long it took to stop?

Reply:
Yes, I can use the Duration.

Reply:
Fixed.